lib/ain-ocean/src/indexer/oracle.rs

```diff
@@ -39,11 +44,202 @@ impl Index for UpdateOracle {
 }
 
 impl Index for SetOracleData {
-    fn index(&self, services: &Arc<Services>, ctx: &Context) -> Result<()> {
-        todo!()
+    fn index(&self, services: &Arc<Services>, context: &Context) -> Result<()> {
+        let feeds = map_price_feeds(vec![self], vec![context])?;
+        let mut pairs: HashSet<(String, String)> = HashSet::new();
+        for feed in feeds {
+            pairs.insert((feed.token.clone(), feed.currency.clone()));
+            services.oracle_price_feed.by_id.put(&feed.id, &feed)?;
+        }
+
+        for (token, currency) in pairs.iter() {
+            let aggregated_value = map_price_aggregated(services, context, token, currency);
+
+            if let Some(value) = aggregated_value {
+                let aggreated_id = (
+                    value.token.clone(),
+                    value.currency.clone(),
+                    value.block.height.clone(),
+                );
+                let aggreated_key = (value.token.clone(), value.currency.clone());
+                services
+                    .oracle_price_aggregated
+                    .by_id
+                    .put(&aggreated_id, &value)?;
+                services
+                    .oracle_price_aggregated
+                    .by_key
+                    .put(&aggreated_key, &aggreated_id)?;
+            }
+        }
+        Ok(())
     }
 
     fn invalidate(&self, services: &Arc<Services>, context: &Context) -> Result<()> {
         todo!()
     }
 }
+
+fn map_price_aggregated(
+    services: &Arc<Services>,
+    ctx: &Context,
+    token: &str,
+    currency: &str,
+) -> Option<OraclePriceAggregated> {
+    // Convert Result to Option
```

Comment:
Let's not have it silently fail. Return a `Result<Option<OraclePriceAggregated>>` instead